edi_835_parser/transaction_set/transaction_set.py

```diff
@@ -599,32 +612,33 @@ def serialize_transaction(
 
 		}
 
-		if transaction.provider_adjustment:
-			provider_adjustment_dict.update({
-				'provider_id': transaction.provider_adjustment.provider_id,
-				'edi_transaction_id_st02': transaction.transaction.transaction_set_control_no,
-				'fiscal_period_date': transaction.provider_adjustment.fiscal_period_date,
-				'provider_adjustment_reason_code1': transaction.provider_adjustment.reason_code1,
-				'provider_adjustment_id1': transaction.provider_adjustment.id1,
-				'provider_adjustment_amount1': transaction.provider_adjustment.amount1,
-				'provider_adjustment_reason_code2': transaction.provider_adjustment.reason_code2,
-				'provider_adjustment_id2': transaction.provider_adjustment.id2,
-				'provider_adjustment_amount2': transaction.provider_adjustment.amount2,
-				'provider_adjustment_reason_code3': transaction.provider_adjustment.reason_code3,
-				'provider_adjustment_id3': transaction.provider_adjustment.id3,
-				'provider_adjustment_amount3': transaction.provider_adjustment.amount3,
-				'provider_adjustment_reason_code4': transaction.provider_adjustment.reason_code4,
-				'provider_adjustment_id4': transaction.provider_adjustment.id4,
-				'provider_adjustment_amount4': transaction.provider_adjustment.amount4,
-				'provider_adjustment_reason_code5': transaction.provider_adjustment.reason_code5,
-				'provider_adjustment_id5': transaction.provider_adjustment.id5,
-				'provider_adjustment_amount5': transaction.provider_adjustment.amount5,
-				'provider_adjustment_reason_code6': transaction.provider_adjustment.reason_code6,
-				'provider_adjustment_id6': transaction.provider_adjustment.id6,
-				'provider_adjustment_amount6': transaction.provider_adjustment.amount6
+		if transaction.provider_adjustments:
+			for provider_adjustment in transaction.provider_adjustments:
```

Comment:
if we are iterating through to update the `provider_adjustment_dict`, wouldn't this just override the provider_adjustment dictionary? why would we want to do this overriding?

Reply:
Before we just have a if statement that checks `if transaction.provider_adjustment:` however the ClaimLoop no longer only has one `provider_adjustment`, so this actually throws an error when I was running the unit tests. Now we store multiple provider adjustments, I am not sure which one should we take? @lrpauley

Reply:
these `serialize_...` methods are meant to consolidate a transaction down to a single pandas dataframe. for this kind of nested data structure, it's hard to flatten a claim down to a single record. overwriting the adjustments like this will roughly preserve the original functionality of these methods (which used to just look at one of the adjustments) and let the unit tests pass, so that should be fine

Reply:
got it, thanks!